shielder/README.md

```diff
@@ -0,0 +1,136 @@
+* a Shielder contract instance (its address is also a part of `deploy/accounts.json`)
+* two accounts: Damian and Hans, with addresses `5D34dL5prEUaGNQtPPZ3yN5Y6BnkfXunKXXz6fo7ZJbLwRRH` and `5GBNeWRhZc2jXu7D55rBimKYDk8PGk8itRYFTPfC8RJLKG5o`, respectively (inspect the `deploy/deploy.sh` if the ones pasted here don't work for you).
+* there's also a third account, Alice, which has admin rights. That said, for the purpose of this tutorial we'll stick to using the regular accounts.
+
+### Showing assets
+
+We can view what's inside the shielder (however, note that this gives you virtually no information about who actually put what there). In order to do that, we will need to use the `shielder-cli`. The easiest way to obtain it is to get it from the `cli/target/release/` folder, like so:
+```bash
+cd cli
+./target/release/shielder-cli show-assets
+```
+
+The output is not exactly fascinating right now, because we haven't deposited any assets yet. It will contain a line like:
+```
+assets=[]
+```
+
+In order to make it more interesting, we need to deposit some assets: let's!
+
+### Depositing assets
+
+Depositing some PSP22 tokens will be only slightly more involved. We need to run the following command:
+```bash
+./target/release/shielder-cli deposit 0 10
+```
+
+This will deposit 10 of the first token into the Shielder (we have two PSP22 tokens registered with id-s 0 and 1).
+It will ask you for the seed of the account depositing the tokens. Type `//0` for Damian and `//1` for Hans (and press 'Enter').
+
+Now, running `show-assets` should be slightly more interesting and show you something like this:
+```
+assets=[Asset { token_id: 0, token_amount: 10, deposit_id: 0 }]
+```
+Which seems to make sense, given that we've just deposited 10 _tokens_ into the Shielder.
+
+To make it more interesting, let's deposit more tokens, first as Damian (typing `//0` as the seed):
+```bash
+./target/release/shielder-cli deposit 0 25
+```
+Which yields:
+```
+assets=[Asset { token_id: 0, token_amount: 35, deposit_id: 0 }]
+```
+(note how the notes have been merged into one).
+
+And later as Hans (`//1`):
+```bash
+./target/release/shielder-cli deposit 0 33
+```
+
+Which in turn yields:
+```
+assets=[Asset { token_id: 0, token_amount: 68, deposit_id: 0 }]
+```
+
+An interesting takeaway from those trivial experiments is that even by inspecting what's inside the Shielder, we can't really say anything useful about the state of ownership.
+Of course, you can try depositing the second token (1) and see a second entry created in the assets lists.
+
+A natural course of action after checking out the deposists would be to check out the withdrawals. Let's do just that!
+
+### Interacting with PSP22 contracts
+
+Before we proceed with our withdrawals, let's figure out a way to inspect the balances of our PSP22 tokens. There are two ways of doing that: the Contracts UI and the `cargo contracts` CLI. We will use the former on account of the simplicity and user-friendliness.
```

Comment:
`on account of the simplicity and user-friendliness -> on account of the simplicity and user-friendliness of the process`